### PR TITLE
Add grid version validation

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/32177

## Description
added a validation for grid- typed fields to be fromversion >= 5.5.0

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 * Fixed an issue where an irrelevant id_set validation ran in the **validate** command when using the *--id-set* flag.
 * Fixed an issue were **generate-docs** command has failed if a command did not exist in commands permissions file.
 * Improved a **validate** command message for missing release notes of api module dependencies.
+* Added a validation for invalid fromVersion field in indicators of Grid type in **validate** command.
```
Comment (bakatzir) [inline CHANGELOG.md:5]:
```suggestion
* Added a validation for the version of incident field of type grid.
```

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 2]
```diff
@@ -356,3 +358,15 @@ def is_changed_type(self):
                     is_type_changed = True
 
         return is_type_changed
+
+    def is_valid_grid_fromversion(self):
+        # type: () -> bool
+        """Validate that a incident field with type grid is from version > 5.5.0"""
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/incident_field.py:364]:
```suggestion
        """Validate that a incident field of type grid is from version > 5.5.0"""
```

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 3]
```diff
@@ -356,3 +358,15 @@ def is_changed_type(self):
...
+        if self.current_file.get('type') == 'grid':
+            if LooseVersion(self.current_file.get('fromVersion', '0.0.0')) < LooseVersion('5.5.0'):
+                error_message, error_code = Errors.no_minimal_fromversion_in_file('fromVersion', '5.5.0')
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/incident_field.py:367]: Please add a new proper error in the errors.py file, e.g: "has a from version of... but the proper from version is 5.5.0"
```suggestion
                error_message, error_code = Errors.incident_field_type_grid_minimal_version(current_file_version, '5.5.0')
```

**Comment by bakatzir** [comment]
Please also run your newly written validation on the content repo and report here the findings.

**Comment by JasBeilin** [comment]
After running the validation on all content, these files are invalid:

Packs/Compliance/IncidentFields/incidentfield-residentsemailaddress.json - [IF112]
Packs/Compliance/IncidentFields/incidentfield-affectedindividualscontactinformation.json - [IF112]
Packs/Malware/IncidentFields/incidentfield-Infected_Hosts.json - [IF112]
Packs/Pcysys/IncidentFields/incidentfield-Pentera_Operation_Details.json - [IF112]
Packs/Smokescreen_IllusionBLACK/IncidentFields/incidentfield-IllusionBLACK_Events.json - [IF112]
Packs/Code42/IncidentFields/incidentfield-Code42_File_Events.json - [IF112]
Packs/CortexXDR/IncidentFields/XDR_File_Artifacts.json - [IF112]
Packs/CortexXDR/IncidentFields/XDR_Network_Artifacts.json - [IF112]
Packs/CortexXDR/IncidentFields/XDR_Alerts.json - [IF112]
Packs/PcapAnalysis/IncidentFields/incidentfield-PCAP_Flows.json - [IF112]
Packs/PrismaCloud/IncidentFields/incidentfield-Prisma_Cloud_Rules.json - [IF112]
Packs/Claroty/IncidentFields/incidentfield-Claroty_Related_Assets.json - [IF112]
Packs/ExtraHop/IncidentFields/incidentfield-ExtraHop_Detection_Participants.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Removed_Parents.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Targets.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Removed_Permissions.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Restriction_Changes.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Added_Permissions.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Added_Parents.json - [IF112]
Packs/GoogleDrive/IncidentFields/incidentfield-Google_Drive_Activity_Mentioned_Users.json - [IF112]
Packs/Phishing/IncidentFields/incidentfield-emailheaders.json - [IF112]
Packs/CommonTypes/IncidentFields/incidentfield-criticalassets.json - [IF112]
Packs/CommonTypes/IncidentFields/incidentfield-urlsslverification.json - [IF112]

**Comment by bakatzir** [comment]
@JasBeilin So what happens when you spin up a 5.0.0 demisto and try to install it? please try.
@Bargenish FYI.

**Comment by bakatzir** [comment]
After talking with Genish and Nitzan,  this validation is not needed as `grid` type was introduced at 5.0.0
and all new released content is fromversion 5.0.0 anyhow.

**Pull request closed without merging**